sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslJoinTest.java

```diff
@@ -230,7 +224,9 @@ public void testRejectsUnboundedWithinWindowsWithEndOfWindowTrigger() throws Exc
                    .accumulatingFiredPanes());
     PCollectionTuple inputs = tuple("ORDER_DETAILS1", orders, "ORDER_DETAILS2", orders);
 
-    thrown.expectCause(expectedSingleFireTrigger());
+    thrown.expect(UnsupportedOperationException.class);
+    thrown.expectMessage(
+        stringContainsInOrder(Arrays.asList("once per window", "default trigger")));
```

Comment:
<!--new_thread; commit:a207bbffa47dfdd01562ef114379df669bf247bf; resolved:1-->
I'm 50/50 on these. I do hold to the discipline that a test should not be factored or engineered, but should be readable in isolation as pretty much straight line. At the same time, this one seemed pretty benign. If it were written as a matcher it would probably pass muster.